### PR TITLE
CI: test on more JDKs

### PR DESCRIPTION
**Pull request by vhotspur**
Added more environments to be tested on GitHub Actions CI.

It now runs on

 * Linux: all JDKs from 8 to 16
 * Linux with OpenJ9: 8, 11, 16
 * MacOS: 8, 11, 13, 15
 * Windows: 8, 11, 13, 15

Also added `--add-opens` to JMH runs for JDK 16 (see #241).

### CODE REVIEW

File: tools/ci/common.sh [hunk 1]
```diff
@@ -21,6 +21,12 @@ RENAISSANCE_GIT_VERSION=$(git describe --tags --always --dirty=-SNAPSHOT || echo
 # Strip leading 'v' from the git-produced version
 RENAISSANCE_VERSION=${RENAISSANCE_GIT_VERSION#v}
 
+# Try to guess JVM version
+RENAISSANCE_JVM_MAJOR_VERSION="$( java -XshowSettings:properties -version 2>&1 \
+    | sed -n -e 's#^[ \t]*java.version[ ]*=[ ]\(.*\)#\1#p' \
```
Comment (lbulej) [inline tools/ci/common.sh:26]: I would write the `sed` expression as follows:
```
sed -n '/^[[:blank:]]*java[.]version[[:blank:]]*=/ { s/.*=[[:blank:]]*\([^[:space:]]\+\)/\1/; s/^1[.]//; s/^\([^.]\+\).*/\1/; p }'
```
but it's your call. I'm a bit uncomfortable with the `#` characters in the `sed` expression (my brain sees comments).

Also, is it necessary to use the `-X` option? The following works with just the `-version` output on JDK 8, 11, 16, and Hotspot 15. On OpenJ9 11, it gets the opendjk version from the first line, which is probably fine too:
```
sed -n '/version[[:blank:]]\+/ { s/.*version[[:blank:]]\+"\([^[:space:]"]\+\)/\1/; s/^1[.]//; s/^\([^.]\+\).*/\1/; p }'
```

If we really want to use the `-X` option, why not ask for the `java.specification.version`? The `sed` expression only needs to strip `1.` from the `1.8` version (you could even keep it there).
```
 sed -n '/^[[:blank:]]*java[.]specification[.]version[[:blank:]]*=/ { s/.*=[[:blank:]]*\([^[:space:]]\+\)/\1/; s/^1[.]//; p }'
```
This worked on OpenJDK 8, 11, 16, Hotspot 15, and OpenJ9 11.

File: tools/ci/bench-jmh.sh [hunk 2]
```diff
@@ -6,6 +6,7 @@ source "$(dirname "$0")/common.sh"
 
 java -jar "$RENAISSANCE_JMH_JAR" \
 	-jvmArgs -Xms2500M -jvmArgs -Xmx2500m \
+	$( for arg in $( get_jvm_workaround_args ); do echo "-jvmArgs" "$arg"; done ) \
```
Comment (lbulej) [inline tools/ci/bench-jmh.sh:9]: Unless we put other workaround args in here, I'd suggest to just name it `get_jvm_add_opens_args` for now.

**Review by lbulej — APPROVED** [review]
LGTM

File: tools/ci/common.sh (hunk 1 above)
Comment (vhotspur) [reply tools/ci/common.sh]: Okay, using `java -version` (hopefully they will not unify it with `--version` that has slightly different output) and used `/` as delimiter in `sed` ;-) [tested on JDK 8 to 16 including OpenJ9].

File: tools/ci/bench-jmh.sh (hunk 2 above)
Comment (vhotspur) [reply tools/ci/bench-jmh.sh:9]: I would say that the fact that it adds only `--add-opens` is an implementation detail from the point of view of the wrapper script.

**Comment by lbulej** [comment]
LEBTM :-)

**Comment by ceresek** [comment]
That's just one step away from LoBoToMy :-)

**Pull request merged**